src/network.js

```diff
@@ -222,14 +224,14 @@ function writeReadMessage (conn, msg, callback) {
         return callback(err)
       }
       if (res.length === 0) {
-        return callback(new Error('No message received'))
+        return callback(errcode(new Error('No message received'), 'ERR_NO_MESSAGE_RECEIVED'))
       }
 
       let response
       try {
         response = Message.deserialize(res[0])
       } catch (err) {
-        return callback(new Error('failed to deserialize response: ' + err.message))
+        return callback(errcode(new Error(`failed to deserialize response: ${err.message}`), 'ERR_FAILED_DESERIALIZE_RESPONSE'))
```

Comment:
There's a few places where we're losing the call stack by recreating the error. Since we're using error codes now, I think losing the nicer message is worth keeping the call stack. 
```suggestion
        return callback(errcode(err, 'ERR_FAILED_DESERIALIZE_RESPONSE'))
```